### PR TITLE
Feature/quick edit lead

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

File: src/typings/lead.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    source: number;
+    sourceRaw?: string;
+    sourceDetail: OrganizationDetail;
+    athors?: [number];
+    authorRaw?: string;
+    authorsDetail: [OrganizationDetail];
+
+    text?: string;
+    url?: string;
+    website?: string;
```
Comment (tnagorra) [inline src/typings/lead.tsx:51]: These should be made exlusive

File: src/typings/lead.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    priorityDisplay: string;
+    project: number;
+    publishedOn: string;
+    status: LeadStatus;
+    sourceType: string;
+
+    source: number;
+    sourceRaw?: string;
+    sourceDetail: OrganizationDetail;
+    athors?: [number];
```
Comment (tnagorra) [inline src/typings/lead.tsx]: This should be a typo
```suggestion
    authors?: [number];
```

File: src/typings/lead.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    assessmentId?: number;
+    assignee?: number;
+    assigneeDetails: UserDetails;
+    confidentiality: Confidentiality;
+    priority: number;
+    priorityDisplay: string;
+    project: number;
+    publishedOn: string;
+    status: LeadStatus;
+    sourceType: string;
```
Comment (tnagorra) [inline src/typings/lead.tsx]: Is sourceType not a Enum?

File: src/typings/lead.tsx (hunk 1 above)
Comment (AdityaKhatri) [reply src/typings/lead.tsx:51]: Oh okay.
Could you take care of this when you are working on AddLeads?

File: src/typings/lead.tsx (hunk 3 above)
Comment (AdityaKhatri) [reply src/typings/lead.tsx]: I'll check

File: src/views/LeadAdd/LeadDetail/index.js [hunk 4]
```diff
@@ -713,6 +719,8 @@ class LeadDetail extends React.PureComponent {
                     pending: pendingSearchedOrganizations,
                 },
             },
+            hideProjects,
+            disableLeadIdentity,
```
Comment (tnagorra) [inline src/views/LeadAdd/LeadDetail/index.js]: What does this mean? Kappa Syata!

File: src/views/Leads/ActionButtons/LeadEditModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        lead: leadFromProps,
+        closeModal,
+        requests: {
+            leadEditRequest: {
+                do: onLeadEdit,
+                pending,
+            },
+        },
+    } = props;
+    console.warn('here', leadFromProps);
```
Comment (tnagorra) [inline src/views/Leads/ActionButtons/LeadEditModal/index.tsx]: Let's not write console.warn on render

Comment (tnagorra) [reply src/views/Leads/ActionButtons/LeadEditModal/index.tsx]: :smiling_imp:

File: src/views/Leads/ActionButtons/LeadEditModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+        faramValues: newFaramValues,
+        faramErrors: newFaramErrors,
+    }) => {
+        setLeadFaramValues(newFaramValues);
+        setLeadFaramErrors(newFaramErrors);
+        setPristine(false);
+    }, [setLeadFaramValues, setLeadFaramErrors, setPristine]);
+
+    const handleFormSubmit = useCallback(() => {
+        detachedFaram({
```
Comment (tnagorra) [inline src/views/Leads/ActionButtons/LeadEditModal/index.tsx:116]: Not a fan of using detachedFaram
I think we should include `Submit` inside lead details conditionally.

File: src/views/Leads/ActionButtons/LeadEditModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+    const lead = useMemo(() => ({
+        faramValues: leadFaramValues,
+        faramErrors: leadFaramErrors,
+    }), [leadFaramValues, leadFaramErrors]);
+
+    const isSubmitDisabled = useMemo(() => (
+        !doesObjectHaveNoData(leadFaramErrors) || pristine
+    ), [pristine, leadFaramErrors]);
+
```
Comment (tnagorra) [inline src/views/Leads/ActionButtons/LeadEditModal/index.tsx:144]: There are certain cases when this should not be done.
I forgot the cases though

File: src/typings/lead.tsx (hunk 1 above)
Comment (tnagorra) [reply src/typings/lead.tsx:51]: Maybe!

File: src/views/Leads/ActionButtons/LeadEditModal/index.tsx (hunk 6 above)
Comment (AdityaKhatri) [reply src/views/Leads/ActionButtons/LeadEditModal/index.tsx:116]: Lets keep it as is for now.

File: src/typings/lead.tsx (hunk 3 above)
Comment (tnagorra) [reply src/typings/lead.tsx]:
```python
    SOURCE_TYPES = (
        (TEXT, 'Text'),
        (DISK, 'Disk'),
        (WEBSITE, 'Website'),
        (DROPBOX, 'Dropbox'),
        (GOOGLE_DRIVE, 'Google Drive'),

        (RSS, 'RSS Feed'),
        (EMM, 'EMM'),
        (WEB_API, 'Web API'),
        (UNKNOWN, 'Unknown'),
    )
```

**Review by tnagorra — APPROVED** [review]

**Pull request merged**